docs/middleware.md

```diff
@@ -0,0 +1,109 @@
+## Adding a middleware on pallet-ibc
+
+### Overview
+
+The middleware is a module that wraps another module and adds some additional functionality to it. In the context of
+IBC, it means that the middleware is a module that is executed before/after the IBC module, and can access the packet
+data, take fees, send new packets etc. The middleware should be considered not a separate module, but rather a part of
+the module that
+it's wrapping.
```

Comment:
so it's just one big module, right?

Reply:
Yes, just with multiple layers, I would say